Lib/enum.py

```diff
@@ -414,6 +500,46 @@ def __new__(metacls, cls, bases, classdict, **kwds):
             if _order_ != enum_class._member_names_:
                 raise TypeError('member order does not match _order_')
         #
+        # remove Flag structures if final class is not a Flag
+        if (
+                Flag is None and cls != 'Flag'
+                or Flag is not None and not issubclass(enum_class, Flag)
+            ):
+            delattr(enum_class, '_boundary_')
+            delattr(enum_class, '_flag_mask_')
+            delattr(enum_class, '_all_bits_')
+            delattr(enum_class, '_inverted_')
+        elif Flag is not None and issubclass(enum_class, Flag):
+            # ensure _all_bits_ is correct and there are no missing flags
+            single_bit_total = 0
+            multi_bit_total = 0
+            for flag in enum_class._member_map_.values():
+                if _is_single_bit(flag._value_):
+                    single_bit_total |= flag._value_
+                else:
+                    # multi-bit flags are considered aliases
+                    multi_bit_total |= flag._value_
+            if enum_class._boundary_ is not KEEP:
+                # missed_bits = multi_bit_total & ~single_bit_total
+                missed = list(_iter_bits_lsb(multi_bit_total & ~single_bit_total))
+                # while missed_bits:
+                #     i = 2 ** (missed_bits.bit_length() - 1)
+                #     missed.append(i)
+                #     missed_bits &= ~i
+                if missed:
+                    raise TypeError(
+                            'invalid Flag %r -- missing values: %s'
+                            % (cls, ', '.join((str(i) for i in missed)))
+                            )
+            enum_class._flag_mask_ = single_bit_total
+            #
+            # set correct __iter__
+            inverted = ~enum_class(0)
+            if list(enum_class) != list(inverted):
+                if '|' in inverted._name_:
```

Comment:
perhaps `!is_single_bit(inverted.value)` is more explicit

Reply:
I don't think that would work -- I'm trying to discern between a brand new combination of all flags, versus an in-the-class alias of all flags -- `WHITE` in `Color`, for example.

Reply:
Noise removed.  I'd like to get this committed by the weekend.